### ISSUE TITLE
Merge reading of drag and drop notifications into core.

### ISSUE THREAD

**Issue by TheQuinbox** [josephsl/wintenApps#75]
If I don't have this add-on installed and use alt+shift+_left and right in the Windows 11 22H2 start menu, I get no speech. If I install the add-on, I get speech exactly as Narrator announces it. It would be amazing if this could be merged into core.

**Comment by josephsl** [comment]
This calls for finishing nvaccess/nvda#12271.

**Comment by TheQuinbox** [comment]
Is something blocking you from doing that? It would be extremely useful.

**Comment by josephsl** [comment]
Hi,

Blocked by: implementation details (specifically on handling focus changes) and life priorities. At one point I did create a pull request about this, but after assessing implementation details, coupled with graduate school work, I decided to let it go. I'm sure there are other parts of this add-on that ought to be in NVDA Core, but those should be discussed in separate issues.

Thanks.

**Comment by TheQuinbox** [comment]
Hi,
I've personally found no issues with how it works in this add-on, so what makes it more difficult to merge into core? I was able to rearrange my start menu tiles in order to add stuff to folders fully well with this add-on, and had no issues.
Thanks.

**Comment by josephsl** [comment]
Hi,

The whole thing consists of three parts:

1. Recognizing drag events: NVDA does not recognize drag complete, drop target dropped, and drag drop effect events and properties. Therefore the add-on adds them manually at startup.
2. Announcing drag and drop properties and events: the base implementation announces the result of drag and drop operations.
3. Handling focus changes: in some cases, dragging and dropping things can cause focus to change and/or cause something other than the element being dragged to announce results.

Of these, the pull request I worked on was stalled with item 3. I and @LeonardDeR had a talk about drag and drop user experience while reviewing my PR.

Thanks.

**Comment by josephsl** [comment]
Relevant PR: nvaccess/nvda#12428

**Comment by josephsl** [comment]
Hi,

Resolved, now part of NVDA 2022.4 with mitigations in place in Windows App Essentials add-on to support old and upcoming NVDA releases.

Thanks.

**Comment by TheQuinbox** [comment]
Awesome, thanks so much for your work on this!
